### ISSUE TITLE
[PageHeader] Button id's are not passing accessibility checking

### ISSUE THREAD

**Issue by youngjunekoh** [carbon-design-system/ibm-products#3543]
follow-up issue from office hours 10/4

## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

This issue is a follow up from office hours 10/4

When adding `id`s to pageActions buttons, accessibility checker is failing for duplicate id's.
![Screenshot 2023-10-04 at 11 45 32 AM](https://github.com/carbon-design-system/ibm-products/assets/60585804/62f84e44-12aa-4aa0-8c3b-73d43220f044)
![Screenshot 2023-10-04 at 11 45 26 AM](https://github.com/carbon-design-system/ibm-products/assets/60585804/7f03ee03-41a1-4cbf-999e-20ddcf9c19f0)

was not able to recreate this in the storybook because the pageActions in the storybook do not have id's specified.

Thank you!

> Is this issue related to a specific component?
PageHeader (with pageAction buttons)

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using? 
v2.10.2 (latest)

**Comment by matthewgallo** [comment]
https://github.com/carbon-design-system/ibm-products/pull/3545
